common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -470,6 +527,39 @@ void closeAndDeleteOutdatedPartitions(
       });
   }
 
+  void deleteMergedFiles(AppAttemptShuffleMergeId appAttemptShuffleMergeId, int[] reduceIds) {
+    removeAppShufflePartitionInfoFromDB(appAttemptShuffleMergeId);
+    AppShuffleInfo appShuffleInfo = validateAndGetAppShuffleInfo(appAttemptShuffleMergeId.appId);
```

Comment:
This call is happening in a threadpool and can be delayed - it is possible that appShuffleInfo was updated to a  different attempt id by this time.
Pass this from the caller.

Reply:
Thanks for your review.
But for this method, shuffle merge files only related to appId, shuffleId, shuffleMergeId, reduceId. so it also make sense with a different attempt id？

Reply:
Pass `appShuffleInfo` to `deleteMergedFiles` method to avoid unnecessary method call.